x-pack/plugins/fleet/server/services/epm/packages/package_verification.ts

```diff
@@ -35,8 +36,53 @@ export async function _readGpgKey(): Promise<string | undefined> {
     logger.warn(`Unable to retrieve GPG key from '${gpgKeyPath}': ${e.code}`);
     return undefined;
   }
+  let key;
+  try {
+    key = await openpgp.readKey({ armoredKey: buffer.toString() });
+  } catch (e) {
+    logger.warn(`Unable to parse GPG key from '${gpgKeyPath}': ${e}`);
+  }
 
-  const key = buffer.toString();
-  cachedKey = key;
   return key;
 }
+
+export interface PackageVerificationResult {
+  verified: boolean;
+  keyId: string;
+}
+
+export async function verifyPackageSignature({
```

Comment:
I've tested this function extensively locally, however to unit test it I would need .zip, .zip.sig and a gpg key file locally, which seemed out of scope of a unit test. The other thing which made me lean towards not unit testing is that this function is a thin wrapper around the openpgp library.

Reply:
+1 I don't think there's much value on unit testing this specific method as it's not adding any logic on top of the `openpgp` package. We're just calling a few methods provided by the library and chaining their results.